datalad/resources/website/assets/js/main.js

```diff
@@ -0,0 +1,222 @@
+ * @param {string} url current url
+ * @param {string} node_name clicked node name
+ * @return {string} url to parent of current url
+ */
+function parent_url(url, node_name) {
+  var turl = url.charAt(url.length - 1) === '/' ? url : url.concat("/");
+  var url_array = turl.split(/[\\/]/);
+  url_array.splice(-2, 1);
+  return url_array.join('/');
+}
+
+/**
+ * construct child path from current url and clicked node name
+ * @param {string} url current url
+ * @param {string} node_name clicked node name
+ * @return {string} url to reach clicked node
+ */
+function child_url(url, node_name) {
+  var turl = url.charAt(url.length - 1) === '/' ? url.slice(0, -1) : url;
+  var tnode_name = node_name.charAt(0) === '/' ? node_name.slice(1) : node_name;
+  return turl + '/' + tnode_name;
+}
+
+/**
+ * if path given return path else return window.location.pathname
+ * replaces direct calls to window.location with function
+ * allows mocking tests for functions using window.location.pathname
+ * @return {string} returns path to current window location
+ */
+function loc() {
+  return window.location;
+}
+
+/**
+ * decompose url to actual path to node
+ * e.g if next_url = d1/d2/d3, current_url = example.com/ds/?dir=d1/d2
+ * return example.com/ds/d1/d2/d3
+ * @param {string} next_url name of GET parameter to extract value from
+ * @return {string} returns path to node based on current location
+ */
+function absolute_url(next_url) {
+  if (!next_url)
+    return loc().pathname;
+  else
+    return (loc().pathname.replace(/\?.*/g, '') + next_url).replace('//', '/');
+}
+
+/**
+ * extract GET parameters from URL
+ * @param {string} name name of GET parameter to extract value from
+ * @param {string} url url to extract parameter from
+ * @return {string} returns the value associated with the `name` GET parameter if exists else null
+ */
+function getParameterByName(name, url) {
+// refer https://stackoverflow.com/questions/901115/how-can-i-get-query-string-values-in-javascript
+  if (!url) url = loc().href;
+  name = name.replace(/[\[\]]/g, "\\$&");
+  var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)");
+  var results = regex.exec(url);
+  if (!results || !results[2]) return null;
+  return decodeURIComponent(results[2].replace(/\+/g, " "));
+}
+
+/**
+ * update url parameter or url ?
+ * @param {string} next_url next url to traverse to
+ * @param {string} type type of clicked node
+ * @param {string} current_state current node type. (variable unused)
+ * @return {boolean} true if clicked node not root dataset
+ */
+function update_param_or_path(next_url, type, current_state) {
+  // if url = root path(wrt index.html) then append index.html to url
+  // allows non-root dataset dirs to have index.html
+  // ease constrain on non-datalad index.html presence in dataset
+  if (next_url === loc().pathname || next_url === '/' || !next_url) {
+    return false;
+  } else if (type === 'file' || type === 'link')
+    return false;
+  else
```

Comment:
Just a style comment, feel free to ignore. But I've never liked the shortcut of omitting of curly braces for one-line if/elses in JS. And here you're mixing both styles side-by-side. It visually implies the flow is different, even though, to my eye, it isn't.